src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaTokenService.cs

```diff
@@ -39,34 +44,24 @@ public string AddTokenToPath(string path)
         {
             var pathIndex = path.IndexOf('?');
 
-            var parsed = pathIndex != -1
-                ? QueryHelpers.ParseQuery(path.Substring(pathIndex + 1))
-                : null;
+            Dictionary<string, StringValues> processingCommands = null;
+            Dictionary<string, StringValues> otherCommands = null;
 
-            // If no commands or only a version command don't bother tokenizing.
-            if (parsed is null || parsed.Count == 0 || parsed.Count == 1 && parsed.ContainsKey(ImageVersionProcessor.VersionCommand))
+            if (pathIndex != -1)
             {
-                return path;
+                ParseQuery(path.Substring(pathIndex + 1), out processingCommands, out otherCommands);
             }
 
-            var processingCommands = new Dictionary<string, string>(parsed.Count);
-            Dictionary<string, string> otherCommands = null;
-
-            foreach (var command in parsed)
+            // If no commands or only a version command don't bother tokenizing.
+            if (processingCommands is null
+                || processingCommands.Count == 0
+                || processingCommands.Count == 1 && processingCommands.ContainsKey(ImageVersionProcessor.VersionCommand))
             {
-                if (_knownCommands.Contains(command.Key))
-                {
-                    processingCommands[command.Key] = command.Value.ToString();
-                }
-                else
-                {
-                    otherCommands ??= new Dictionary<string, string>();
-                    otherCommands[command.Key] = command.Value.ToString();
-                }
+                return path;
             }
 
             // Using the command values as a key retrieve from cache
-            var queryStringTokenKey = CreateQueryStringTokenKey(processingCommands.Values);
+            var queryStringTokenKey = CreateQueryStringTokenKey(processingCommands);
```

Comment:
`.Values` allocates new `ValueCollection`, no need for that so let's just pass the dictionary which has efficient enumeration